controllers/chat/getSignedChannelList.js

```diff
@@ -14,11 +22,21 @@ const __queryChannelBuilder = async (userId, limit, offset) => {
   return channels;
 };
 
+/**
+ *
+ * @param {import('stream-chat').Channel} channel
+ * @returns
+ */
 const __transformChannelData = (channel) => {
   const newChannel = {...channel.data};
   delete newChannel.config;
   delete newChannel.own_capabilities;
 
+  const messageLength = channel.state.messages?.length;
+  const channelType = newChannel?.channel_type;
+
+  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return;
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__transformChannelData` has been modified to include a check for the channel type and message length. If the message length is 0 and the channel type is 'TOPIC', the function returns undefined. This could potentially lead to an array with undefined elements when this function is used in a map operation. Consider filtering out these channels before transforming them, or handle the undefined values where this function is called.
```diff
- if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return;
+ if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return null;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
already changed to reduce function